resources/js/components/GlobalSearch.vue

```diff
@@ -0,0 +1,282 @@
+            paginationPeople: 0,
+            peopleFlag: true,
+            locationList: [],
+            locationCount: 0,
+            paginationLocation: 0,
+            locationFlag: true,
+            debounce: null
+        };
+    },
+    computed: {
+        navigationList () {
+            const list = [];
+            if (this.topicsList.length) list.push('Themen');
+            if (this.locationList.length) list.push('Orte');
+            if (this.peopleList.length) list.push('Personen');
+            return ['Alle', ...list];
+        },
+        isNothing () {
+            return this.inputValue &&
+                !this.loading &&
+                !this.topicsList.length &&
+                !this.locationList.length &&
+                !this.peopleList.length;
+        }
+    },
+    methods: {
+        focusHandle (value) {
+            this.onFocus = value;
+            if (value) {
+                this.$refs.globalSearchInput.focus();
+                document.body.style.overflow = 'hidden';
+            } else {
+                document.body.style.overflow = '';
+            }
+        },
+        async getTopics () {
+            if (!this.topicsFlag) return;
+            this.paginationTopics += 1;
+            const { data, meta } = await topics.getTopicsLike(this.inputValue, this.paginationTopics);
+            this.topicsCount = meta.total;
+            this.topicsList = [...this.topicsList, ...data];
+            this.loading = false;
+            if (!data.length) this.topicsFlag = false;
+
+        },
+        async getPeople () {
+            if (!this.peopleFlag) return;
+            this.paginationPeople += 1;
+            const { members, meta } = await people.getPeopleLike(this.inputValue, this.paginationPeople);
+            this.peopleCount = meta.total;
+            this.peopleList = [...this.peopleList, ...members];
+            this.loading = false;
+            if (!members.length) this.peopleFlag = false;
+        },
+        async getLocations () {
+            if (!this.locationFlag) return;
+            this.paginationLocation += 1;
+            const { locations, meta } = await location.getLocationLike(this.inputValue, this.paginationLocation);
+            this.locationCount = meta.total;
+            this.locationList = [...this.locationList, ...locations];
+            this.loading = false;
+            if (!locations.length) this.locationFlag = false;
+        },
+        lazyHandle () {
+            this.debounce && clearTimeout(this.debounce);
+            this.debounce = setTimeout(
+                () => {
+                    if (this.activeTab === 'Themen') this.getTopics();
+                    if (this.activeTab === 'Orte') this.getLocations();
+                    if (this.activeTab === 'Personen') this.getPeople();
+                }, 500);
+        },
+        changeTab (item) {
+            this.activeTab = item;
+            setTimeout(() => this.createObserver(), 0);
```

Comment:
why do you need this? if there's any edge case that you need to do it this way then please leave a comment